---
layout: review
title: "Highly accurate protein structure prediction with AlphaFold"
tags: protein-folding biology deep-learning
author: "Robin Trombetta"
cite:
    authors: "John Jumper, Richard Evans, Alexander Pritzel et al."
    title:   "Highly accurate protein structure prediction with AlphaFold"
    venue: " Nature"
pdf: "https://www.nature.com/articles/s41586-021-03819-2#citeas"
---

# Highlights

* AlphaFold2 is an end-to-end deep learning network for prediction the 3D structure of proteins from the sequence of amino acids.
* The model won the 14th Critical Assessment of protein Structure Prediction (CASP14)  challenge in 2020 by a large margin, acheving performances never seen before.
* The work of this team was recently awarded by a Nobel Price in chemistry (awarded to John Jumper and Demis Hassabis from Google DeepMind, as well as David Becker) for having "cracked the code for proteins' amazing structure".
* The paper, published in Nature, is rather short by itself, but all the details about the model, the training procedure and more are available in the 62-page-long [supplementary material](https://static-content.springer.com/esm/art%3A10.1038%2Fs41586-021-03819-2/MediaObjects/41586_2021_3819_MOESM1_ESM.pdf).
* [This video](https://www.youtube.com/watch?v=ri39B0Voujc) is of great use to understand in detail AlphaFold2.

&nbsp;

# Introduction

Understanding the 3D structure of proteins, these key building blocks of life on Earth, are essential for biologists. The sequence of amino acids constituing a protein is not sufficient to know its function, which is also greatly defined by the spatial arrangement of the molecule. Obtained the 3D structure of a protein is feasible experimentally but is also incredibly costly and long : until now, the structures of around 100 000 unique proteins have been determined experimentally. However, predicting the 3D structure of a protein from its 1D sequence of amino acids remained until few years an unsolved task. Such task is of major interest as it has the potential to greatly accelerate the research in biology and chemistry.

In 2020, Google DeepMind's team called AlphaFold2 won the bi-annual CASP14 challlenge by a large margin over the other methods and achieved unprecedented accuracy, in par with experimental eror. Two years before, DeepMind had already won the challenge but the performance of the first version of their model was not as groundbreaking as its second iteration.

<div style="text-align:center">
<img src="/collections/images/AlphaFold/intro.jpg" width=400></div>
<p style="text-align: center;font-style:italic">Figure 1. Example of the 3D structure of a protein obtained experimentally (green) and predicted by AlphaFold (blue).</p>

&nbsp;

**Brief introduction to protein structure**. A protein is a molecule composed of one or more polypeptides, that is linear chains of *amino acid residues*. Shorter protein only contain only fews dozens of residues but the longer ones can have thousands of amino acids. An amino acid is composed of "main structure", or *backbone* of two atoms of carbons, an amin group, a carboxyl group and a *side chain*, often called R (Figure 2). The different amino acids of a protein are bonded by a link between the amino and the carboxyl groups.

<div style="text-align:center">
<img src="/collections/images/AlphaFold/amino_acid.jpg" width=200></div>
<p style="text-align: center;font-style:italic">Figure 2. General structure of an alpha-amino acids composing proteins.</p>

&nbsp;

# Motivation

The task solved by AlphaFold is quite far from our domain of reseach, image processing, but I do believe that reviewing this paper can be of interest for us, and not only for general scientific culture purpose.

AlphaFold2 is not only a feat of engineering, it also contains significant methodological contributions, and can be an inspiration for us in how we build deep learning based models for medical images, in particular showing the interest of adding appropriate inductive biases into the models.


# Overview of AlphaFold 

<div style="text-align:center">
<img src="/collections/images/AlphaFold/overview.jpg" width=800></div>
<p style="text-align: center;font-style:italic">Figure 3. High-level overview of AlphaFold.</p>

Let's first see how AlphaFold works on a high level (Figure 3). It takes as input a 1D sequence of amino acids, more specifically of the 22 $$\alpha$$-amino acids that are present in proteins, and it aims to output the full 3D structure of the molecule, that is, the position and orientation in space of all its atoms. It combines two complementary approaches that have a long history in protein structure prediction that focus respectively on the physical interactions or the evolutionary history of a protein. Understanding the physical interactions heavily between the atoms can greatly guide the prediction of the structure of proteins via thermodynamic or/and kinetic simulations. However, these methods are usually computationally intractable and require accurately model the physic of the interaction inside the protein. On the other hand, evoolutionary approaches leverage the history of proteins, homologies between proteins of different species and pairwise cevolutionary correlations. It is allowed by the existance of the millions of protein sequences already available in the Protein Data Bank.

## Input embedding

From an input amino acid sequence, AlphaFold derives two main objets that will be used to predict the 3D structure : the Multi Sequence Alignment (MSA) matrix between the protein of interest and proteins with similar function on different species and the pairwise representation between elements of the input sequence, embedded with features from template proteins (Figure 4).

The idea behind using the MSA matrix is to analyze the mutations that has occured inside amino acid sequence of proteins that share a common ancestor ad the same function. For instance, intuitively, when comparing the proteins of two species, if two amino acids always mutate at the same time, this can means that there are really closeby in the 3D structure of the protein.


<div style="text-align:center">
<img src="/collections/images/AlphaFold/input_embedding.jpg" width=800></div>
<p style="text-align: center;font-style:italic">Figure 4. AlphaFold input embedding into MSA features and pairwise representation.</p>


&nbsp;


# Deep dive into the architecture of AlphaFold


&nbsp;




# Reference

[^1]: C. Ryali, Y.-T. Hu, D. Bolya, C. Wei, H. Fan, P.-Y. Huang, V Aggarwal, A. Chowdhury, O. Poursaeed, J. Hoffman, J. Malik, Y. Li, and C. Feichtenhofer. Hiera: A hierarchical vision transformer without the bells-and-whistles. International Conference on Machine Learning (ICML), 2023.

